synapse/handlers/room.py

```diff
@@ -802,7 +813,7 @@ async def send(etype, content, **kwargs) -> int:
                 _,
                 last_stream_id,
             ) = await self.event_creation_handler.create_and_send_nonmember_event(
-                creator, event, ratelimit=False
+                creator, event, ratelimit=False, ignore_shadow_ban=True,
```

Comment:
why do we ignore the shadow-ban here? comment please.

Reply:
also: do we not need to ignore it for the `update_membership` call below?

Reply:
I believe that we need to ignore it here so that the creation happens successfully -- @erikjohnston does that sound correct?

I suspect we need to ignore it in the call below too. I'll add a test for creating rooms.

Reply:
Oh wait, we don't need it in the call below because the changes to `update_membership` only affect *invites*. I'll add some comments.

Reply:
Check out 5adcc56c8f9d6d38657c21f3634f143c2b29b004